tests/context.py

```diff
@@ -116,33 +116,69 @@ def getPythonCommand():
 		>>>
 
 		Testcase 1: function should have a output.
-		
-		>>> tests.context.getPythonCommand() is not None
+
+		>>> tests.context.getCoverageCommand() is not None
 		True
 		>>>
 
+
 	"""
-	thepython = "exit 1 ; #"
+	thecov = "exit 1 ; #"
 	try:
-		thepython = checkPythonCommand(["command", "-v", "coverage"])
-		if (sys.version_info >= (3, 3)):
-			if (str("/coverage") in str(thepython)) and (sys.version_info >= (3, 3)):
-				thepython = str("coverage run -p")
-			elif str("/coverage3") in str(checkPythonCommand(["command", "-v", "coverage3"])):
-					thepython = str("coverage3 run -p")
-			else:
-				thepython = str(sys.executable)
-		elif (str("/coverage") in str(thepython)) and (sys.version_info <= (3, 2)):
-			try:
-				import coverage
-				if coverage.__name__ is not None:
-					thepython = str("{} -m coverage run -p").format(str(sys.executable))
-				else:
-					thepython = str(sys.executable)
-			except Exception:
-				thepython = str(sys.executable)
+		thecov = checkPythonCommand(["command", "-v", "coverage"])
+		if (str("/coverage") in str(thecov)):
+			thecov = str("coverage")
+		elif str("/coverage3") in str(checkPythonCommand(["command", "-v", "coverage3"])):
+			thecov = str("coverage3")
 		else:
+			thecov = "exit 1 ; #"
+	except Exception:
+		thecov = "exit 1 ; #"
+	return str(thecov)
+
+
+def __check_cov_before_py():
+	"""
+		Utility Function to check for coverage before just python.
+	"""
+	thepython = str(sys.executable)
+	thecov = getCoverageCommand()
+	if (sys.version_info >= (3, 3)):
+		if (str("coverage") in str(thecov)) and (sys.version_info >= (3, 3)):
+			thecov += str(" run -p")
+	elif (str("coverage") in str(thecov)) and (sys.version_info <= (3, 2)):
```

Comment:
```suggestion
	if (str("coverage") in str(thecov)) and (sys.version_info >= (3, 7)):
	 	 thecov += str(" run -p")
	 	 thepython = str("{} -m {}").format(str(sys.executable), str(thecov))
	else:
```
complexity without reason is an ant-pattern